nRF Toolbox/Profiles/Cycling/Sections/CyclingTableViewSection.swift

```diff
@@ -0,0 +1,158 @@
+    private var cadence: Int = 0
+    
+    var items: [DefaultDetailsTableViewCellModel] = [
+        DefaultDetailsTableViewCellModel(title: "Speed", identifier: .speed),
+        DefaultDetailsTableViewCellModel(title: "Cadence", identifier: .cadence),
+        DefaultDetailsTableViewCellModel(title: "Distance", identifier: .distance),
+        DefaultDetailsTableViewCellModel(title: "Total Distance", identifier: .totalDistance),
+        DefaultDetailsTableViewCellModel(title: "Gear Retio", identifier: .gearRatio)
+    ]
+    
+    func dequeCell(for index: Int, from tableView: UITableView) -> UITableViewCell {
+        let detailsCell = tableView.dequeueCell(ofType: DetailsTableViewCell.self)
+        detailsCell.update(with: items[index])
+        return detailsCell
+    }
+    
+    var numberOfItems: Int { return items.count }
+    
+    let sectionTitle: String = "Speed and Cadence"
+    
+    var id: Identifier<Section> = .cycling
+    
+    private func update(_ item: DefaultDetailsTableViewCellModel) -> DefaultDetailsTableViewCellModel {
+        var item = item
+        
+        let speeedFormatter = MeasurementFormatter.numeric(maximumFractionDigits: 1)
+        let distanceFormatter = MeasurementFormatter.numeric(maximumFractionDigits: 2)
+        
+        switch item.identifier {
+        case .speed:
+            item.value = speeedFormatter.string(from: speed)
+        case .distance:
+            item.value = distanceFormatter.string(from: travelDistance)
+        case .totalDistance:
+            item.value = distanceFormatter.string(from: totalTravelDistance)
+        case .cadence:
+            item.value = "\(cadence) RPM"
+        case .gearRatio:
+            item.value = String(format: "%.2f", gearRatio)
+        default:
+            break
+        }
+        return item
+    }
+    
+    mutating func update(with data: Data) {
+        let value = UnsafeMutablePointer<UInt8>(mutating: (data as NSData).bytes.bindMemory(to: UInt8.self, capacity: data.count))
+        
+        let flag = value[0]
+        
+        var wheelRevDiff: Double = 0
+        var crankRevDiff: Double = 0
+        
+        if flag & wheelRevolutionFlag == 1 {
+            wheelRevDiff = processWheelData(value)
+            if flag & 0x02 == 2 {
+                crankRevDiff = processCrankData(value, revolutionIndex: 7)
+                if crankRevDiff > 0 {
+                    gearRatio = wheelRevDiff / crankRevDiff
+                }
+            }
+        } else if flag & crankRevolutionFlag == 2 {
+            crankRevDiff = processCrankData(value, revolutionIndex: 1)
+            if crankRevDiff > 0 {
+                gearRatio = wheelRevDiff / crankRevDiff
+            }
+        }
+        
+        items = items.map(self.update)
+    }
+    
+    private mutating func processWheelData(_ value: UnsafeMutablePointer<UInt8>) -> Double {
+        let wheelRevolution = UInt8(CFSwapInt32LittleToHost(UInt32(value[1])))
```

Comment:
This seems to be super too complicated.

Reply:
Please, get rid of UnsafeMutablePointers :)
Use https://github.com/NordicSemiconductor/IOS-nRF-Mesh-Library/blob/version2/nRFMeshProvision/Classes/Type%20Extensions/Data.swift instead.

Reply:
Updated processing code. 
Moved part of the code to special model.
Rid of `UnsafeMemoryPointer`s